src/engine/Board.java

```diff
@@ -484,23 +486,44 @@ public Zombie getSingleAirTarget(int x, int y) {
 	 * Deletes all the units within the specified row
 	 * @param row - the row that all the units will be deleted in
 	 */
-	public void useLawnMower(int row)
+	public List<Zombie> useLawnMower(int row)
 	{
-		for(Unit u : getRowUnits(row))
+		List<Unit> unitRemoveBin = getRowUnits(row);
+		List<Zombie> zomRemoveBin = new LinkedList<Zombie>();
+		
+		if(!unitRemoveBin.isEmpty())
```

Comment:
You  don't need to check if its empty or not

Reply:
Yeah, true. I suppose everytime it is called, there must always be at least one zombie on the board. Even if there were no zombies on the board I should have checked if unitRemoveBin was null.